arangod/VocBase/LogicalCollection.h

```diff
@@ -88,6 +88,21 @@ class LogicalCollection : public LogicalDataSource {
 
   enum class Version { v30 = 5, v31 = 6, v33 = 7, v34 = 8, v37 = 9 };
 
+  /*
+   * @brief Describes a more specific collection type
+   * then document / edge that is implicitly generated by the features
+   * of this collection and not explicitly created by the user.
+   * Mostly used for Enterprise/Smart collection types which have
+   * some specialities over community collections.
+   * For Backwards Compatibility a Value can never be reused
+   */
+  enum class InternalCollectionType {
+    Community = 1,
+    SmartEdge = 2,
+    LocalSmartEdge = 3,
+    RemoteSmartEdge = 4
+  };
```

Comment:
I am not a huge fan of this, because with this a collection now has a `_type` and an `_internalCollectionType`.
It would be good to fuse them into just one. I understand this will be a lot of work and affects all drivers etc., so I understand using a second internal variable for it.
However, as far as I can see it is only used for edge collections, so it should be named appropriately, e.g. `_edgeCollectionType`. I also think `Community` is not a good name for the non-smart edge collections, because these collections can also be created with the EE. It would rename it to `Normal`, even if that sounds boring.

Reply:
How about `Standard` instead of `Normal`?

Reply:
The reason why I did this is to not expose this non-modifyable value to the user, it is of pure internal use only.
Having this exposed limits us in what we can do with it (see drivers) having this not-exposed (and separated from document/edge type) gives us freedom to only be backwards compatible by 1 version, or even erase it at all if we do not need it anymore.

I a open to store this value into the "old" type attribute internally and translate it before we send it out to the user, to not change the drivers. But I do not know if it is worth it.

I am happy to rename the first to `Standard` or `Normal`.

Also we may want to introduce e.g. "SmartJoin" type here as well, as this collections also requires the same speciality as the SmartEdges, so.i am against moving it to an edge only type, this are just the first entries here.